frontend/src/metabase/dashboard/hoc/DashboardControls.jsx

```diff
@@ -213,6 +216,29 @@ export default ComposedComponent =>
       };
 
       render() {
+        const {
+          dashboard,
+          parameterValues,
+          draftParameterValues,
+          setParameterValues,
+        } = this.props;
+        const hasUnappliedFilters =
+          isNotNull(draftParameterValues) &&
+          !_.isEqual(parameterValues, draftParameterValues);
+
+        const handleApplyFilters = () => {
+          setParameterValues(draftParameterValues);
+        };
+
+        const applyFilterButton = !dashboard?.auto_apply_filters ? (
```

Comment:
I opt to pass the whole `applyFilterButton` rather than passing a bunch of individual props and construct the button inside the view to keep related logic in the same place.

This component `DashboardControls` is shared among public dashboards and app dashboards.

Reply:
Could you explain why decided to use this approach instead of rendering the button directly in the place it's supposed to be and "connect"-ing the button to not pass all the required props? This approach with passing ReactNode's around is not something we almost never do

Reply:
You don't need to pass new props. You should be able to call `connect` or `useSelector` / `useDispatch` at the very bottom. WDYT?